### PR TITLE
chore(deps): weekly dependency updates

### PR DESCRIPTION
**Pull request by ironyh**
# Pull Request

## Description

Weekly dependency maintenance to keep the project's Node/Vue/tooling ecosystem current. This PR applies safe version bumps, regenerates the lockfile, and enforces a single `@types/node` version across all packages to ensure consistent type-checking.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (code improvements without changing functionality)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Test improvements
- [ ] Build/CI/CD changes
- [x] Dependencies update

## Related Issues

Related: #

## What's Changed

- Bumped devDependencies in `package.json` (Playwright, `@types/node`, TypeScript ESLint, Vue, Typedoc plugin, Wrangler)
- Refreshed `pnpm-lock.yaml` to reflect updated dependency graph and transitive bumps
- Added `"@types/node": "^22"` to `pnpm.overrides` to force a single `@types/node` major across all workspace packages (prevents transitive deps like `@types/qrcode` from pulling in `@types/node@25.x`)

### Unit Tests
- [ ] Added/updated unit tests for new/changed functionality
- [x] All unit tests pass (`pnpm test`)
- [ ] Test coverage maintained or improved

**Unit Test Results:**
```
Test Files  253 passed (253)
     Tests  7729 passed | 1 skipped (7730)
```

### Integration Tests
- [ ] Added/updated integration tests where applicable
- [ ] All integration tests pass (`pnpm test:integration`)
- [ ] Edge cases covered

### Manual Testing
- [ ] Tested in development environment
- [ ] Tested with real SIP server/provider
- [ ] Tested on multiple browsers (if UI changes)
- [ ] Tested on mobile devices (if applicable)
- [x] Verified no console errors or warnings

## Code Quality Checklist

- [ ] TypeScript types are properly defined (no `any` unless absolutely necessary)
- [x] Code follows project style guide and conventions
- [ ] ESLint passes with no errors (`pnpm lint`)
- [ ] No ESLint suppressions added (or justified in comments if necessary)
- [x] Code is DRY (Don't Repeat Yourself) - no unnecessary duplication
- [ ] Functions/methods have single responsibility
- [ ] Public APIs have JSDoc documentation
- [ ] Complex logic includes inline comments
- [ ] No memory leaks (event listeners cleaned up, timeouts cleared)
- [ ] Error handling is comprehensive and user-friendly
- [ ] Accessibility considerations addressed (if UI changes)

## Breaking Changes

- [ ] This PR includes breaking changes

**Breaking changes details:**
None.

## Screenshots/Demos

N/A — no UI changes.

## Performance Impact

- [x] No significant performance impact

## Documentation

- [ ] README.md updated (if applicable)
- [ ] User guides updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] Code comments added/updated for complex logic
- [ ] CHANGELOG.md updated (if applicable)

## Reviewer Notes

**Audit Results:**
```
No known vulnerabilities found
```

The `pnpm.overrides` entry `"@types/node": "^22"` ensures all packages — including those that previously resolved `@types/node@25.5.2` (e.g. `@types/qrcode`) — now consistently use `@types/node@22.19.17`, matching the root devDependency and the Node 20/22 CI target.

## Checklist Before Requesting Review

- [x] Code is self-reviewed
- [ ] Changes follow the project's contribution guidelines (see [CONTRIBUTING.md](../CONTRIBUTING.md))
- [x] Commit messages are clear and follow project conventions
- [ ] Branch is up to date with base branch
- [x] No merge conflicts
- [ ] All CI checks pass

---

**For Reviewers:**

Please verify:
- [ ] Code quality and style consistency
- [ ] Test coverage is adequate
- [ ] Documentation is clear and complete
- [ ] No security vulnerabilities introduced
- [ ] TypeScript types are properly defined
- [ ] Changes align with project architecture and goals

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -7103,38 +7103,37 @@ snapshots:
...
     dependencies:
-      '@types/node': 22.19.15
+      '@types/node': 22.19.17
       pg-protocol: 1.13.0
       pg-types: 2.2.0
 
   '@types/qrcode@1.5.6':
     dependencies:
-      '@types/node': 22.19.15
+      '@types/node': 25.5.2
```
[the remark this answers is not in the document]
Comment (ironyh) [reply pnpm-lock.yaml:7221]: @copilot apply changes based on [this feedback](https://github.com/ironyh/VueSIP/pull/213#discussion_r3037792216)
